addon/components/basic-dropdown.js

```diff
@@ -89,10 +90,23 @@ export default @layout(templateLayout) @tagName('') class BasicDropdown extends
     if (this.publicAPI.disabled || this.publicAPI.isOpen) {
       return;
     }
-    if (this.onOpen && this.onOpen(this.publicAPI, e) === false) {
+    if (this.onOpen) {
+      deprecate("onOpen is deprecated and will be removed in the future. Please use onBeforeOpen instead", false, {
+        id: 'ember-basic-dropdown.onOpen',
+        until: '3.0.0',
+        url: 'https://ember-basic-dropdown.com/docs/dropdown-events',
+      });
+      if (this.onOpen(this.publicAPI, e) === false) {
+        return;
+      }
+    }
+    if (this.onBeforeOpen && this.onBeforeOpen(this.publicAPI, e) === false) {
       return;
     }
     this.updateState({ isOpen: true });
+    if (this.onAfterOpen) {
+      this.onAfterOpen(this.publicAPI, e);
```

Comment:
I'm not 100% if we should call it here or wait until the `afterRender` step. 
I lean towards your solution, and let the user wait for the render if they want to.